### PR TITLE
Support HEAD requests to the raw API

### PR DESCRIPTION
**Pull request by chrismwendt**
This allows language servers to quickly determine whether or not the given repository exists on the Sourcegraph instance without needing to construct a Sourcegraph-specific GraphQL query. cc @felixfbecker this might be useful for TypeScript too.

See https://github.com/sourcegraph/go-langserver/pull/345#discussion_r240768252

### CODE REVIEW

File: cmd/frontend/internal/app/ui/raw.go [hunk 1]
```diff
@@ -124,6 +126,17 @@ func serveRaw(w http.ResponseWriter, r *http.Request) error {
 		if relativePath == "" {
 			relativePath = "."
 		}
+
+		if r.Method == "HEAD" {
```
Comment (felixfbecker) [inline cmd/frontend/internal/app/ui/raw.go]: Why support HEAD only if the content type is tar or zip?

**Comment by felixfbecker** [comment]
The raw API uses Sourcegraph-specific repo names, so don't you still need to first resolve the git clone URL to a Sourcegraph repo name?

File: cmd/frontend/internal/app/ui/raw.go (hunk 1 above)
Comment (chrismwendt) [reply cmd/frontend/internal/app/ui/raw.go]: No reason, I'll make this work for any content type.

**Comment by chrismwendt** [comment]
In the go-langserver it's not necessary to resolve the clone URL to a Sourcegraph repo name because this is a performance optimization https://github.com/sourcegraph/go-langserver/pull/345#discussion_r240794386

> I plan to implement option 1 above with the assumption that the repo name is the same as the clone URL, which is usually true. When it's not, the raw API will likely 404 and go-langserver will fall back to a git clone.

**Comment by slimsag** [comment]
This doesn't support HEAD requests for directories / files _within_ a repository but that's okay for now (especially given you don't need that)

File: cmd/frontend/internal/app/router/router.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func newRouter() *mux.Router {
 	repo.Path("/badge.svg").Methods("GET").Name(RepoBadge)
 
 	// Must come last
-	base.PathPrefix("/").Methods("GET").Name(UI)
+	base.PathPrefix("/").Name(UI)
```
Comment (felixfbecker) [inline cmd/frontend/internal/app/router/router.go]: What would happen now if I do a POST? (I would expect a 405)

Comment (felixfbecker) [reply cmd/frontend/internal/app/router/router.go]: nvm I see there is a subrouter further down

**Pull request merged**